src/blocks/header/mod.rs

```diff
@@ -424,6 +334,17 @@ impl BlockHeader {
         //   signature encoded.
         fvm_ipld_encoding::to_vec(&blk).expect("block serialization cannot fail")
     }
+
+    /// Fetch a block header from the blockstore. This call fails if the header
+    /// is present but invalid. If the header is missing, None is returned.
+    pub fn load(store: impl Blockstore, key: Cid) -> anyhow::Result<Option<BlockHeader>> {
```

Comment:
Should `&impl Blockstore` be sufficient?

Reply:
It's up to the caller whether they want to pass ownership.